macros/src/main/scala/MacroCompiler.scala

```diff
@@ -677,18 +678,39 @@ object MacroCompiler extends App {
         // Run the compiler.
         val result = new MacroCompiler().compileAndEmit(state)
 
-        // Extract Verilog circuit and write it.
-        verilogWriter.write(result.getEmittedCircuit.value)
-      }
+        // Write output FIRRTL file.
+        params.get(Firrtl) match {
+          case Some(firrtlFile: String) => {
+            val fileWriter = new FileWriter(new File(firrtlFile))
+            fileWriter.write(result.circuit.serialize)
+            fileWriter.close()
+          }
+          case None =>
+        }
+
+        // Write output Verilog file.
+        params.get(Verilog) match {
+          case Some(verilogFile: String) => {
+            // Open the writer for the output Verilog file.
+            val verilogWriter = new FileWriter(new File(verilogFile))
 
-      // Close the writer.
-      verilogWriter.close()
+            // Extract Verilog circuit and write it.
+            verilogWriter.write(result.getEmittedCircuit.value)
+
+            // Close the writer.
+            verilogWriter.close()
+          }
+          case None =>
```

Comment:
Since this isn't optional shouldn't you print an nice error message here? Or is this just going to fallback to printing the usage?

Reply:
The idea is that you could choose firrtl output or Verilog output. Should I add a warning neither option is used?

Reply:
Oh I see. I don't think a neither check is necessary.